addons/arsenal/functions/fnc_attributeInit.sqf

```diff
@@ -24,14 +24,12 @@ if (_mode > 0) then {
     // Blacklist: add full arsenal and take items away
     [_object, true, true] call FUNC(initBox);
 
-    // Need to delay removal by 2 frames
+    // Wait until all items have been added, so that the blacklisted items can be removed
     [{
-        [{
-            params ["_object", "_items"];
-
-            [_object, _items, true] call FUNC(removeVirtualItems);
-        }, _this] call CBA_fnc_execNextFrame;
-    }, [_object, _items]] call CBA_fnc_execNextFrame;
+        !isNil {(_this select 0) getVariable QGVAR(virtualItems)}
+    }, {
+        [_this select 0, _this select 1, true] call FUNC(removeVirtualItems);
+    }, [_object, _items], 20] call CBA_fnc_waitUntilAndExecute; // 20s timeout in case of failure
```

Comment:
🔥

Reply:
Not sure if you are saying it's lit or if it should be set on fire...

Reply:
![the-road-to-el-dorado-both](https://github.com/acemod/ACE3/assets/69561145/fd365909-7e7e-4c0b-80ec-9aacd9223783)